util/ranger/ranger_test.go

```diff
@@ -1193,7 +1193,7 @@ func (s *testRangerSuite) TestIndexStringIsTrueRange(c *C) {
 	testKit.MustExec("drop table if exists t0")
 	testKit.MustExec("CREATE TABLE t0(c0 TEXT(10));")
 	testKit.MustExec("INSERT INTO t0(c0) VALUES (1);")
-	testKit.MustExec("CREATE INDEX i0 ON t0(c0(10));")
+	testKit.MustExec("CREATE INDEX i0 ON t0(c0(255));")
```

Comment:
why change this?

Reply:
Thanks for your comment!
When I run `make test` after completing the code for converting, this test case shows tidb doesn't use index when the length doesn't equal to the type length.
I just commit this to look for someone to help me to pass the test.

Reply:
it may be caused by you changed the `col.FieldType.Flen` when you change `col.FieldType.Tp`

Reply:
But when I modify the length to 9 based on the code from pingcap/master and then run `make test`, it still logs as follows:
```
... obtained string = "" +
...     "[TableReader_7 0.00 root  data:Selection_6]\n" +
...     "[└─Selection_6 0.00 cop[tikv]  gt(test.t0.c0, \"123\"), test.t0.c0]\n" +
...     "[  └─TableFullScan_5 1.00 cop[tikv] table:t0 keep order:false]\n"
... expected string = "" +
...     "[IndexReader_7 1.00 root  index:Selection_6]\n" +
...     "[└─Selection_6 1.00 cop[tikv]  test.t0.c0]\n" +
...     "[  └─IndexRangeScan_5 1.00 cop[tikv] table:t0, index:i0(c0) range:(\"123\",+inf], keep order:false]\n"
... sql:explain select * from t0 where c0 and c0 > '123', args:[]
```
By the way, did I misunderstand the conversion? Should I change the length?

Reply:
@Abingcbc hi, I'm sorry for my fault,  after I test on MySQL, `col.FieldType.Flen` should be changed when blob type is automatically converted.  

> this test case shows tidb doesn't use index when the length doesn't equal to the type length.

I will find out why this happened, you can change this test case to pass the test.

Reply:
hi, I have confirmed that tidb doesn't use index when the length doesn't equal to the type length.  it's ok to change this test.